### PR TITLE
Misssing collectors cleanup

### LINKED ISSUE #70: Technical debt collectors from add testing PR

### PR DESCRIPTION
**Pull request by waltsims**
This PR will add missing collectors to test suite. Will close #70.

### CODE REVIEW

**Review by faridyagubbayli — APPROVED** [review]

File: kwave/utils/filters.py [hunk 1]
```diff
@@ -233,37 +233,63 @@ def extract_amp_phase(data: np.ndarray, Fs: float, source_freq: float, dim: Tupl
...
+
+        # compute metric
+        s = (convolve(im, sobel3D) ** 2 + convolve(im, np.transpose(sobel3D, (2, 0, 1))) ** 2 +
+             convolve(im, np.transpose(sobel3D, (1, 2, 0))) ** 2).sum()
+    return s
 
-    else:
-        raise ValueError("Invalid number of dimensions in im")
-    return sobel.sum()
+    # NOTE: Walter thinks this is the proper way to do this, but it doesn't match the MATLAB version
```
Comment (faridyagubbayli) [inline kwave/utils/filters.py:270]: Looks neat, I vote for implementing Walter's version :)

File: kwave/utils/matrix.py [hunk 2]
```diff
@@ -119,7 +111,6 @@ def resize(mat: Union[np.ndarray, List[List[Union[int, float]]]], new_size: Unio
     else:
         mat_rs = mat_rs.reshape(new_size, order='F')
     # update command line status
-    print(f'  completed in {scale_time(TicToc.toc())}')
```
Comment (faridyagubbayli) [inline kwave/utils/matrix.py:122]: Why to remove the timing info?

File: kwave/utils/signals.py [hunk 3]
```diff
@@ -410,6 +410,7 @@ def tone_burst(sample_freq, signal_freq, num_cycles, envelope='Gaussian', plot_s
     else:
         signal = np.zeros([tone_index.size, signal_length])
 
+    # TODO (walter): Logic of signal length shorter than signal not covered
```
Comment (faridyagubbayli) [inline kwave/utils/signals.py:413]: Should this be addressed now or later?

File: tests/matlab_test_data_collectors/matlab_collectors/collect_brenner.m [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+img2 = rand([5,5]);
+out2 = sharpness(img2, 'Brenner');
+recorder.recordVariable('img2', img2);
+recorder.recordVariable('out2', out2);
+
+img3 = rand([5,5,5]);
+out3 = sharpness(img3, 'Brenner');
+recorder.recordVariable('img3', img3);
+recorder.recordVariable('out3', out3);
+recorder.saveRecordsToDisk();
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_brenner.m:13]: missing new line at the end of file

File: tests/matlab_test_data_collectors/matlab_collectors/collect_createCWSignals.m [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+phase = linspace(0, 2*pi, 9).';
+signal = createCWSignals(t_array, f, amp, phase);
+  
+recorder.recordVariable('t_array', t_array);
+recorder.recordVariable('f', f);
+recorder.recordVariable('amp', amp);
+recorder.recordVariable('phase', phase);
+recorder.recordVariable('signal', signal);
+
+recorder.saveRecordsToDisk();
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_createCWSignals.m:22]: missing new line at the end of the file

File: tests/matlab_test_data_collectors/matlab_collectors/collect_gaussian.m [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+
+x = (-3 : 0.05 : 3);
+
+y = gaussian(x);
+
+recorder.recordVariable('x', x);
+recorder.recordVariable('y', y);
+
+recorder.saveRecordsToDisk();
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_gaussian.m:12]: missing new line at the end of the file

File: tests/matlab_test_data_collectors/matlab_collectors/collect_normvar.m [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+img2 = rand([5,5]);
+out2 = sharpness(img2, 'NormVariance');
+recorder.recordVariable('img2', img2);
+recorder.recordVariable('out2', out2);
+
+img3 = rand([5,5,5]);
+out3 = sharpness(img3, 'NormVariance');
+recorder.recordVariable('img3', img3);
+recorder.recordVariable('out3', out3);
+recorder.saveRecordsToDisk();
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_normvar.m:13]: missing new line at the end of the file

File: tests/matlab_test_data_collectors/matlab_collectors/collect_resize.m [hunk 8]
```diff
@@ -1,49 +1,29 @@
...
-        filename = [output_folder idx_padded '.mat'];
-        save(filename, 'params', 'volume', 'resized_volume' , 'new_size', 'type');
-        k = k + 1;
+    % Save the original volume, resized volume, and interpolation method
+    if ~exist(output_folder, 'dir')
+        mkdir(output_folder);
     end
+    idx_padded = sprintf('%06d', k );
+    filename = [output_folder idx_padded '.mat'];
+    save(filename, 'volume', 'resized_volume', 'new_size', 'method');
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_resize.m:27]: Do you think it is a good time to start using `TestRecorder` here? We can also leave it to some future PR.

File: tests/matlab_test_data_collectors/matlab_collectors/collect_toneBurst.m [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    {80e6, 20e6, 2, 'Envelope', 'Gaussian', 'SignalOffset', 200}, ...
+    {10e6, 0.5e6, 200, 'Envelope', [40, 90], 'SignalLength', 8000} ...
+    {10e6, 0.5e6, 200, 'Envelope', [40, 90], 'SignalLength', 8000, 'SignalOffset', 200} ...
+}; 
+
+for param_idx = 1:length(params)
+    input_signal = toneBurst(params{param_idx}{:});
+    idx_padded = sprintf('%06d', param_idx - 1);
+    filename = ['collectedValues/toneBurst/' idx_padded '.mat'];
+    save(filename, 'params', 'input_signal');
```
Comment (faridyagubbayli) [inline tests/matlab_test_data_collectors/matlab_collectors/collect_toneBurst.m:18]: Maybe use `TestRecorder`?

File: tests/test_utils.py [hunk 10]
```diff
@@ -301,2778 +186,23 @@ def test_gradient_spect_1D():
...
-                    0.415626937777450, 0.475528258147571, 0.523720494614294, 0.559016994374943, 0.580548640463043,
-                    0.587785252292469, 0.580548640463041, 0.559016994374942, 0.523720494614294, 0.475528258147573,
-                    0.415626937777447, 0.345491502812522, 0.266848920427793, 0.181635632001339, 0.0919498715009091,
-                    1.43965865561198e-16, -0.0919498715009091, -0.181635632001338, -0.266848920427793,
-                    -0.345491502812522, -0.415626937777449, -0.475528258147574, -0.523720494614294, -0.559016994374943,
-                    -0.580548640463043, -0.587785252292469, -0.580548640463042, -0.559016994374942, -0.523720494614294,
-                    -0.475528258147573, -0.415626937777450, -0.345491502812522, -0.266848920427793, -0.181635632001338,
-                    -0.0919498715009101, -2.15948798341796e-16, 0.0919498715009096, 0.181635632001338,
-                    0.266848920427791, 0.345491502812522, 0.415626937777449, 0.475528258147574, 0.523720494614294,
-                    0.5590
```
Comment (faridyagubbayli) [inline tests/test_utils.py:205]: Missing todo text

**Review by faridyagubbayli — COMMENTED** [review]
Left some comments

**Comment by waltsims** [comment]
@faridyagubbayli does this pass tests for you locally?

File: kwave/utils/matrix.py (hunk 2 above)
Comment (waltsims) [reply kwave/utils/matrix.py:122]: I added back 📦

File: tests/matlab_test_data_collectors/matlab_collectors/collect_resize.m (hunk 8 above)
Comment (waltsims) [reply tests/matlab_test_data_collectors/matlab_collectors/collect_resize.m:29]: Done

File: kwave/utils/filters.py (hunk 1 above)
Comment (waltsims) [reply kwave/utils/filters.py:270]: in the future...

**Pull request merged**